Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -136,6 +136,36 @@ public void SerializesAndDeserializesCustomPropertyNames()
             Check(TestObjects.fmRadio);
         }
 
+        [TestMethod]
```

Comment:
Why not just use `Check` and pass in an object of a class which has method getters and setters. You can reduce this test to a single line.

Reply:
Because I'm trying to verify that the annotated methods were actually used during serialization/deserialization. Also `Check` could succeed even if the method serialization/deserialization implementation was wrong/absent (via the usual property/field processing).

To do this, I defined annotated getters that return data that differs from the object's internal data (so we can check for this difference on lines 155 & 158) and annotated setters that correct for this difference (we check this on lines 163 & 166).

Also I'm trying to test that we get the proper behavior when we have a naming conflict between an annotated property name and a .NET property/field name (the assert on line 149).